File: WF/WF2003/2003pB.py
def flip(a, pos):
	if a[pos] == "1":
		a[pos] = "0"
	else:
		a[pos] = "1"

cnt = 1
while True:
	a, b = input().split()
	if (a == "0") and (b == "0"):
		break
	if cnt != 1:
		print()
	sa = "{0:b}".format(int(a))
	sb = "{0:b}".format(int(b))
	limit = max(len(sa), len(sb))
	sa = "0" * (limit - len(sa)) + sa
	sb = "0" * (limit - len(sb)) + sb
	lista = []
	listb = []
	lista[:0] = sa
	listb[:0] = sb
	imp = ans = 10e100
	# case 1	
	tmp = 0
	for i in range(1, limit, 1):
		if (lista[i-1] != listb[i-1]):
			tmp += (1 << (limit - i - 1))
			flip(lista, i-1)	
			flip(lista, i)	
			if i + 1 < limit :
				flip(lista, i+1)
	if lista[limit-1] == listb[limit-1] :	
		ans = min(ans, tmp)
#	print("this {} is new ans".format(ans))
	
	tmp = (1 << (limit - 1))
	lista[:0] = sa
	flip(lista, 0)
	flip(lista, 1)
	for i in range(1, limit, 1):
		if (lista[i-1] != listb[i-1]):
			tmp += (1 << (limit - i - 1))
			flip(lista, i-1)	
			flip(lista, i)	
			if i + 1 < limit :
				flip(lista, i+1)
	if lista[limit-1] == listb[limit-1] :	
		ans = min(ans, tmp)
	if(ans != imp):
		print("Case Number {cnt}: {ans}".format(cnt=cnt, ans=ans))
	else:
		print("Case Number {cnt}: {ans}".format(cnt=cnt, ans="impossible"))
	cnt+=1

File: WF/WF2003/2003pB.cpp
#include <bits/stdc++.h>
using namespace std;

#define REP(i, n) for(int (i) = 0 ; (i) < (int)(n); (i)++)
#define REP1(i, n) for(int (i) = 1 ; (i) <= (int)(n); (i)++)
#define yccc cin.tie(0), ios_base::sync_with_stdio(false)
#define PR(x) cout << #x << " = " << (x) << endl
#define endl "\n"
using ll = long long;
using ull = unsigned long long;
using uint = unsigned int;
//bitset<350> sa, sb;



int main(){
//	yccc;
	int cnt = 1;
	string a, b;
	while(cin >> a >> b){
		if(a == "0" && b == "0")
			break;
		//PR(a), PR(b);
		bitset<350> sa(a), sb(b);
		
		//ull pa = a;
		// case 1
		bitset<350> ans;	
		ans.reset();
		ans.flip();
		int limit = 349;
		while(sa[limit] == 0 && sb[limit] == 0)
			limit--;	
		bitset<350> tmp;	
		for(int i = limit-1; i >= 0; i--){
			if(sa[i+1] != sb[i+1]){
				tmp.flip(i);
				sa.flip(i+1), sa.flip(i);
				if(i > 0)
					sa.flip(i-1);
			}
		}
		if(sa[0] == sb[0])
			ans = bitset<350>(min(ans.to_string(), tmp.to_string()));
		// case 2
		sa = bitset<350>(a);
		tmp.reset();
		tmp.flip(limit);
		sa.flip(limit);
		if(limit > 0)
			sa.flip(limit-1);
		for(int i = limit-1; i >= 0; i--){
			if(sa[i+1] != sb[i+1]){
				tmp.flip(i);
				sa.flip(i+1), sa.flip(i);
				if(i > 0)
					sa.flip(i-1);
			}
		}
		if(sa[0] == sb[0])
			ans = bitset<350>(min(ans.to_string(), tmp.to_string()));
// print out ans;
		if(cnt != 1)
			cout << endl;
		cout << "Case Number " << cnt++ << ": ";	
		if(ans == numeric_limits<ll>::max())
			cout << "impossible\n";
		else 
			cout << ans.to_string() << endl;
	}

}
